File: code/backend/php/src/Core/Domain/Auth/TokenInfo.php
<?php

namespace Games\Paths\Core\Domain\Auth;

/**
 * Immutable domain model for parsed JWT claims.
 */
class TokenInfo
{
    private string $userUuid;
    private string $username;
    private string $role;
    private string $type;
    private int $iat;
    private int $exp;
    private ?string $jti;

    public function __construct(
        string $userUuid,
        string $username,
        string $role,
        string $type,
        int $iat,
        int $exp,
        ?string $jti = null
    ) {
        $this->userUuid = $userUuid;
        $this->username = $username;
        $this->role = $role;
        $this->type = $type;
        $this->iat = $iat;
        $this->exp = $exp;
        $this->jti = $jti;
    }

    public function getUserUuid(): string
    {
        return $this->userUuid;
    }

    public function getUsername(): string
    {
        return $this->username;
    }

    public function getRole(): string
    {
        return $this->role;
    }

    public function getType(): string
    {
        return $this->type;
    }

    public function getIat(): int
    {
        return $this->iat;
    }

    public function getExp(): int
    {
        return $this->exp;
    }

    public function getJti(): ?string
    {
        return $this->jti;
    }

    public function isAdmin(): bool
    {
        return $this->role === 'ADMIN';
    }

    public function isAccessToken(): bool
    {
        return $this->type === 'access';
    }

    public function isRefreshToken(): bool
    {
        return $this->type === 'refresh';
    }
}

File: code/backend/java/core/src/main/java/games/paths/core/model/auth/TokenInfo.java
package games.paths.core.model.auth;

/**
 * TokenInfo - Immutable record representing parsed JWT token claims.
 * Used by the authentication filter and session service to carry
 * validated user identity information extracted from a JWT.
 *
 * @param userUuid  user UUID claim
 * @param username  username claim
 * @param role      role claim ({@code PLAYER} / {@code ADMIN})
 * @param type      token type — {@code "access"} or {@code "refresh"}
 * @param tokenId   JWT {@code jti} claim
 * @param issuedAt  issue time, epoch millis
 * @param expiresAt expiry time, epoch millis
 */
public record TokenInfo(
        String userUuid,
        String username,
        String role,
        String type,
        String tokenId,
        long issuedAt,
        long expiresAt) {

    public TokenInfo {
        if (userUuid == null || userUuid.isBlank()) {
            throw new IllegalStateException("userUuid is required");
        }
    }

    public boolean isAccessToken() {
        return "access".equals(type);
    }

    public boolean isRefreshToken() {
        return "refresh".equals(type);
    }

    public boolean isAdmin() {
        return "ADMIN".equals(role);
    }
}
